codegen/module_system.go

```diff
@@ -544,6 +573,57 @@ func (g *httpClientGenerator) Generate(
 	}, nil
 }
 
+// PopulateQPSLevels loops through endpoint dir and gets qps levels
+func PopulateQPSLevels(EndpointsBaseDir string) map[string]int {
+	qpsLevels := make(map[string]int)
+	endpointFiles := GetListOfAllFilesInEndpointDir(EndpointsBaseDir)
+	for _, endpointFile := range endpointFiles {
+		var config map[string]interface{}
+		bytes, _ := ioutil.ReadFile(endpointFile)
+		err := yaml.Unmarshal(bytes, &config)
+		if err != nil {
+			// try json
+			_ = json.Unmarshal(bytes, &config)
+		}
+		clientMethod, methodOK := config["clientMethod"]
+		clientID, clientOK := config["clientId"]
+		qpsLevel, qpsOK := config["qpsLevel"]
+		if methodOK && clientOK && qpsOK && err == nil {
+			// edge case where clientID or clientMethod is nil
+			if clientID != nil && clientMethod != nil && clientID != "" && clientMethod != "" {
```

Comment:
Do we want  || or && checks here?

Reply:
I was thinking && so it only goes in the if block if clientID and clientMethod are valid